web/modules/custom/dpl_opening_hours/src/Plugin/rest/resource/v1/OpeningHoursResource.php

```diff
@@ -88,22 +88,59 @@ public function get(Request $request): Response {
     $typedRequest = new RequestTyped($request);
 
     try {
+      $branchIds = $this->parseBranchIds($typedRequest->getString('branch_id'));
+
       $openingHoursInstances = $this->repository->loadMultiple(
-        $typedRequest->getInt('branch_id'),
+        $branchIds,
         $typedRequest->getDateTime('from_date'),
         $typedRequest->getDateTime('to_date')
       );
 
-      $responseData = array_map(function (OpeningHoursInstance $instance) : OpeningHoursResponse {
+      $responseData = array_map(function (OpeningHoursInstance $instance): OpeningHoursResponse {
         return $this->mapper->toResponse($instance);
       }, $openingHoursInstances);
 
       return (new CacheableResponse($this->serializer->serialize($responseData, $this->serializerFormat($request))))
         ->addCacheableDependency($this->cachableMetadata());
     }
     catch (\TypeError $e) {
-      throw new BadRequestHttpException("Invalid input: {$e->getMessage()}",);
+      throw new BadRequestHttpException("Invalid input: {$e->getMessage()}");
+    }
+  }
+
+  /**
+   * Parses the 'branch_id' parameter into an integer, array(integers), or NULL.
+   *
+   * @param string|null $branchIdParam
+   *   The 'branch_id' query parameter from the request.
+   *
+   * @return int|int[]|null
+   *   Returns an integer if a single valid ID is provided, an array of integers
+   *   if multiple valid IDs are provided, or NULL if no valid IDs are found.
+   */
+  private function parseBranchIds(?string $branchIdParam): int|array|null {
+    if ($branchIdParam === NULL) {
+      return NULL;
     }
+
+    // Split the parameter into an array using commas as delimiters.
+    $branchIdsArray = explode(',', $branchIdParam);
+    // Trim whitespace and filter out empty values.
+    $branchIdsArray = array_filter(array_map('trim', $branchIdsArray), fn($value) => strlen($value) > 0);
```

Comment:
Consider moving the request parameter to array parsing to the `RequestTyped` class.

It seems quite generic/reusable to me if you leave out the non-positive integer part. I do not think it is that important.